presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -131,6 +131,7 @@
 
     private boolean bucketExecutionEnabled = true;
     private boolean sortedWritingEnabled = true;
+    private boolean alwaysGetBucketMetadata = true;
```

Comment:
I kinda don't understand why do we need this property?

If the bucketing is disabled, and the `$bucket` filter is specified, we should just throw an exception with a message that communicates to the used that filtering by `$bucket` is only allowed when bucketed reads are enabled. 

Also it looks like we should remove the extra `isBucketExecutionEnabled` check from the `HiveMetadata#getTableLayouts`. 

Currently bucketing execution will be disabled at the step before (by not setting the BucketHandle).

Reply:
Discussed it offline. No i get it.